src/tabs.ts

```diff
@@ -0,0 +1,258 @@
+            return html`
+${this._animateStyles()}
+<div id="parent" class="parentDiv"
+    style=${this._internal_fashion=='column'?
+            styleMap(fashionColumn):''}>
+    <div id="title-tab" style=
+        "display:flex;
+         flex-direction:${this._internal_fashion=='column'?
+            'column':'row'}">
+        <slot name="tab-title" @click=${this._titleClicked}>
+        </slot>
+    </div>
+    <div id="body-tab">
+        <slot id="body-slot" name="tab-body"></slot>
+    </div>
+</div>
+            `
+        }
+    }
+    override connectedCallback() {
+        super.connectedCallback();
+        this.renderRoot.addEventListener("slotchange", (e) => {
+            this._slotChangedEvent(e)});
+        window.addEventListener("resize", this._resizeListener);
+    }
+    override disconnectedCallback() {
+        window.removeEventListener("resize", this._resizeListener);
+    }
+    override firstUpdated() {
+        if(!this.animation)
+            this.animationDur = 0;
+        this._openActive();
+    }
+    private _slotChangedEvent(event:Event) {
+        if(!event.target)
+            return;
+        let nodes = (event.target as HTMLSlotElement).assignedElements();
+        let name = (event.target as HTMLSlotElement)?.name;
+        if(name == 'tab-body') {
+            for(let i of nodes)
+                if(!i.classList.contains(this.activecls))
+                    (i as HTMLElement).style.display = 'none';
+        } else if(name == 'acc') {
+            for(let i of nodes)
+                if(!i.children[1].classList.contains(this.activecls))
+                    (i.children[1] as HTMLElement).style.display = 'none';
+        }
+    }
+    private _openActive() {
+        let val = this.querySelectorAll("[slot=tab-title]");
+        for(let i = 0;i < val.length; i++) {
+            if(val[i].classList.contains(this.activecls)) {
+                this.show(i);
+                break;
+            }
+        }
+        if(!this._prev_ix)
+            this.show(0);
+    }
+    private _checkFashion():string {
+        let vw = window.innerWidth;
+        for(let style of this._fashion_style)
+            if((style.ll || 0) <= vw && (
+                    !style.ul || style.ul >= vw)) {
+                return style.val;
+            }
+        return '';
+    }
+    private _changeFashionStyle(val:string) {
+        if(this._internal_fashion != val) {
+            if(val == 'accordion')
+                this._showAccordion();
+            else
+                this._showRC();
+            this._internal_fashion = val;
+        }
+    }
+    private _resizeEvent() {
+        let val = this._checkFashion();
+        this._changeFashionStyle(val);
+        this._openActive();
+    }
+    private _showAccordion() {
+        if(this.animation) {
+            let style = document.createElement("style");
+            style.innerHTML = `${animationCSS(
+                '.animation-open', '.animation-close',
```

Comment:
Prem, I beleive these class names can be little bit mnagled, so that they dont clash with the application space. What if there is a .animation-open class already defined, then this definition is bound to collide with it. I suggest we use the class names like .__tab-acc-an-open__ and  .__tab-acc-an-close__. The general programming notion is that anything that starts with __ and ends with it are private variables .